tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -0,0 +1,329 @@
+import * as React from 'react';
+import { Provider } from 'react-redux';
+import { mount } from 'enzyme';
+
+import NotFound from 'amo/components/ErrorPage/NotFound';
+import Link from 'amo/components/Link';
+import UserProfileEdit, {
+  UserProfileEditBase,
+} from 'amo/components/UserProfileEdit';
+import {
+  editUserAccount,
+  fetchUserAccount,
+  getCurrentUser,
+  loadUserAccount,
+} from 'amo/reducers/users';
+import { USERS_EDIT } from 'core/constants';
+import { ErrorHandler } from 'core/errorHandler';
+import ErrorList from 'ui/components/ErrorList';
+import { dispatchSignInActions } from 'tests/unit/amo/helpers';
+import {
+  createFakeEvent,
+  createStubErrorHandler,
+  createUserAccountResponse,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  function defaultUserProps(props = {}) {
+    return {
+      display_name: 'Matt MacTofu',
+      userId: 500,
+      username: 'tofumatt',
+      ...props,
+    };
+  }
+
+  function syncPropsAndParamsUsername(username) {
```

Comment:
I find the name of this function confusing, and I think the whole thing is extra confusing because of the `params` object which I suggested changing above. This function is really doing two things (iiuc):

- set the `username` prop to what I passed in
- log in with the username that I passed in

If that's the case then perhaps it should be given a name that suggests this.